### DISCUSSION TITLE
Getting Model/Table at the state of a revision

### DISCUSSION THREAD

**Discussion by zikphil** [sqlalchemy/alembic#818] [Usage Questions] [answered]
Hello,

I am using Alembic not only to perform DB upgrades but also content migrations and I was wondering if it's possible, WITHIN a migration to get the state of a table or a model using the Alembic state and not my codebase.

Scenario:
```
In my migration, I am removing a field on the User Model and adding a new one.
Also in the migration, I am migrating the old values of that field into it's a new one. In order to do that, I need to import the User model so I can query it.
The problem is that the field no longer exists on my Model since I am deleting it.
```

I want to get the User model at the state at which it was before that particular migration ran. Is it something that's built into alembic or a third-party plugin? My goal is to avoid having to re-write my model inside the migration file, which is kinda messy.

Thanks,

Phil

**Comment by zzzeek** [comment, marked as answer]
you can reflect tables within a migration,  you can use op.get_bind() wtih any reflection-related operation:

    my_table = Table('my_table', MetaData(), autoload_with=op.get_bind())

to have them in the form of "models" you'd need to use automap.

**Reply by zikphil** [reply]
https://docs.sqlalchemy.org/en/latest/orm/extensions/automap.html
